pkg/awsutils/awsutils.go

```diff
@@ -339,21 +339,41 @@ func (cache *EC2InstanceMetadataCache) initWithEC2Metadata() error {
 	}
 	log.Debugf("Found vpc-ipv4-cidr-block: %s ", cache.vpcIPv4CIDR)
 
-	// retrieve vpc-ipv4-cidr-blocks
-	metadataVPCIPv4CIDRs, err := cache.ec2Metadata.GetMetadata(metadataMACPath + mac + metadataVPCcidrs)
-	if err != nil {
-		awsAPIErrInc("GetMetadata", err)
-		log.Errorf("Failed to retrieve vpc-ipv4-cidr-blocks from instance metadata service")
-		return errors.Wrap(err, "get instance metadata: failed to retrieve vpc-ipv4-cidr-block data")
+	// spawing go routine and retrieving vpc-ipv4-cidr-block
+	waitCh := make(chan string)
+	errorCh := make(chan error)
+	go wait.Forever(func() {
+		cache.initVPCIPv4CIDRs(mac, waitCh, errorCh)
+	}, 30 * time.Second)
+
+	select{
+	case <- waitCh:
+		return nil
+	case errs := <- errorCh:
+		return errs
 	}
+}
 
-	vpcIPv4CIDRs := strings.Fields(metadataVPCIPv4CIDRs)
-
-	for _, vpcCIDR := range vpcIPv4CIDRs {
-		log.Debugf("Found VPC CIDR: %s", vpcCIDR)
-		cache.vpcIPv4CIDRs = append(cache.vpcIPv4CIDRs, aws.String(vpcCIDR))
+// initVPCIPv4CIDRs vpc-ipv4-cidr-blocks
+func (cache *EC2InstanceMetadataCache) initVPCIPv4CIDRs(mac string, waitCh chan string, errorCh chan error) {
+	err := retry.RetryNWithBackoff(retry.NewSimpleBackoff(500*time.Millisecond, time.Minute, 0.5, 3), 5, func() error {
```

Comment:
Hmm, the retry is fancy, but for metadata I don't think we need it. We don't use it for any of the other `GetMetadata()` calls, and I can't recall ever seeing it failing on the logs.